pkg/util/ec2/dmi.go

```diff
@@ -0,0 +1,101 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package ec2
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/dmi"
+	"github.com/google/uuid"
+)
+
+func isBoardVendorEc2() bool {
+	if !config.Datadog.GetBool("ec2_use_dmi") {
+		return false
+	}
+
+	vendor := dmi.GetBoardVendor()
+	if vendor == DMIBoardVendor {
+		return true
+	}
+	return false
```

Comment:
Note: Can be written as `return dmi.GetBoardVendor() == DMIBoardVendor`

Reply:
:man_facepalming: